src/main/java/org/truffleruby/core/regexp/TruffleRegexpNodes.java

```diff
@@ -75,7 +75,6 @@
 @CoreModule("Truffle::RegexpOperations")
 public class TruffleRegexpNodes {
 
-    @TruffleBoundary
     public static Matcher createMatcher(RubyContext context, RubyRegexp regexp, Rope stringRope, byte[] stringBytes,
             boolean encodingConversion, int start, Node currentNode) {
```

Comment:
As far as I can see, `encodingConversion` is always true, could you simplify the code based on that?

Reply:
Should I also remove the argument from `regexp_match_in_region`? Or do you mean just within `createMatcher`?

Reply:
Yeah, also for `regexp_match_in_region` (and its callers), otherwise we would ignore it which doesn't sound good.